internal/dwarf/frame/table.go

```diff
@@ -0,0 +1,665 @@
+func ExecuteDwarfProgram(fde *DescriptionEntry) *FrameContext {
+	frameContext := executeCIEInstructions(fde.CIE)
+	frameContext.order = fde.order
+	frame := frameContext.getCurrentInstruction()
+	frame.loc = fde.Begin()
+	// frame.address = pc
+	frameContext.Execute(fde.Instructions)
+	return frameContext
+}
+
+func (frameContext *FrameContext) executeDwarfProgram() {
+	for frameContext.buf.Len() > 0 {
+		executeDwarfInstruction(frameContext)
+	}
+}
+
+// ExecuteUntilPC execute dwarf instructions.
+func (frameContext *FrameContext) ExecuteUntilPC(instructions []byte) {
+	frameContext.buf.Truncate(0)
+	frameContext.buf.Write(instructions)
+
+	// We only need to execute the instructions until
+	// ctx.loc > ctx.address (which is the address we
+	// are currently at in the traced process).
+	frame := frameContext.getCurrentInstruction()
+	// TODO CHANGE
+	for frame.address >= frame.loc && frameContext.buf.Len() > 0 {
+		executeDwarfInstruction(frameContext)
+	}
+}
+
+// Execute execute dwarf instructions.
+func (frameContext *FrameContext) Execute(instructions []byte) {
+	frameContext.buf.Truncate(0)
+	frameContext.buf.Write(instructions)
+
+	// TODO(kakkoyun): Cleanup.
+
+	// We only need to execute the instructions until
+	// ctx.loc > ctx.address (which is the address we
+	// are currently at in the traced process).
+	// for frame.address >= frame.loc &&
+
+	for frameContext.buf.Len() > 0 {
+		/* ins :=  */ executeDwarfInstruction(frameContext)
+		// Just for debugging:
+		// fmt.Fprintf(os.Stderr, "----------- dwarf instruction: %s at index %d\n", CFAString(ins), len(frameContext.instructions))
+	}
+}
+
+func executeDwarfInstruction(frameContext *FrameContext) byte {
+	instruction, err := frameContext.buf.ReadByte()
+	if err != nil {
+		panic("Could not read from instruction buffer")
+	}
+
+	if instruction == DW_CFA_nop {
+		return instruction
+	}
+
+	fn, instruction := lookupFunc(instruction, frameContext.buf)
+	fn(frameContext)
+
+	return instruction
+}
+
+func lookupFunc(instruction byte, buf *bytes.Buffer) (instruction, byte) {
+	const high_2_bits = 0xc0
+	var restore bool
+
+	// Special case the 3 opcodes that have their argument encoded in the opcode itself.
+	switch instruction & high_2_bits {
+	case DW_CFA_advance_loc:
+		instruction = DW_CFA_advance_loc
+		restore = true
+
+	case DW_CFA_offset:
+		instruction = DW_CFA_offset
+		restore = true
+
+	case DW_CFA_restore:
+		instruction = DW_CFA_restore
+		restore = true
+	}
+
+	if restore {
+		// Restore the last byte as it actually contains the argument for the opcode.
+		err := buf.UnreadByte()
+		if err != nil {
+			panic("Could not unread byte")
+		}
+	}
+
+	fn, ok := fnlookup[instruction]
+	if !ok {
+		// msg := fmt.Sprintf("Encountered an unexpected DWARF CFA opcode: %#v", instruction)
+		// fmt.Println(msg)
+		// return unknown
+
+		// TODO(kakkoyun): Why do we have to panic?
+		panic(fmt.Sprintf("Encountered an unexpected DWARF CFA opcode: %#v", instruction))
```

Comment:
This is where we panic if we don't support the opcode @v-thakkar 😄

Reply:
Aha, makes sense. Then we should comment out those instructions until we have a need for the implementation.